src/pages/home/report/ReportActionItemMessage.js

```diff
@@ -40,6 +40,10 @@ function ReportActionItemMessage(props) {
     const styles = useThemeStyles();
     const fragments = _.compact(props.action.previousMessage || props.action.message);
     const isIOUReport = ReportActionsUtils.isMoneyRequestAction(props.action);
+    if (ReportActionsUtils.isChannelLogMemberAction(props.action)) {
+        fragments[0].html = ReportUtils.getChannelLogMemberMessage(props.action, true);
```

Comment:
I don't like this idea of replacing the backend-provided HTML string with a "fixed-up" HTML string.

In my opinion, it would be clearer to do it in this way:

```js
if (ReportActionsUtils.isChannelLogMemberAction(props.action)) {
  return <ChannelLogMemberActionMessage ... />;
}
```

Of course, this isn't the full solution. What input would we pass to this new component?

I think that `getChannelLogMemberMessage` should start returning a "view model" in the form of node list:

```js
[
    {
        kind: 'text',
        content: 'invited ',
    },
    {
        kind: 'userMention',
        userName: 'foo@bar.com',
        userId: 123,
    },
    {
        kind: 'text',
        content: ' and ',
    },
    {
        kind: 'userMention',
        userName: 'foo2@bar.com',
        userId: 124,
    },
    {
        kind: 'text',
        content: ' to ',
    },
    {
        kind: 'roomReference',
        roomName: '#room3',
        roomId: 456,
    },
]
```

We could also provide `getChannelLogMemberMessagePlainText` which does the simple transformation to copyable plain text.

`ChannelLogMemberActionMessage` would perform the view-level transformation of that view model to user-facing components, skipping the textual HTML transformation.

Reply:
Alternatively, `getChannelLogMemberMessage` could take a "presenter" interface instance.

```ts
interface ChannelLogMemberMessagePresenter<Node> {
    presentText(text: String): Node;
    
    presentUserMention(userMention: {
        readonly userName: string,
        readonly userId: number,
    }): Node;

    presentRoomLink(roomLink: {
        readonly roomName: string,
        readonly roomId: number,
    }): Node;
}

const reactChannelLogMemberMessagePresenter: ChannelLogMemberMessagePresenter<ReactNode> = {
    // ...
};

const plainTextChannelLogMemberMessagePresenter: ChannelLogMemberMessagePresenter<string> = {
    // ...
};

function getChannelLogMemberMessage<Node>(
  reportAction,
  presenter: ChannelLogMemberMessagePresenter<Node>,
): ReadonlyArray<Node> {
  // ...
}
```

Reply:
(this is just a bunch of inspirational TypeScript lines, not a full solution sketch)

Reply:
This will be challenging, as we might need to blaze the trail here.